src/prefect/core/flow.py

```diff
@@ -159,6 +159,12 @@ def __init__(
         self.storage = storage
         self.result_handler = result_handler
 
+        if result_handler:
+            warnings.warn(
+                "DEPRECATED: the result_handler Flow option is deprecated, this is being replaced by the prefect.engine.Result object.",
```

Comment:
Maybe now is a good time to decide on the new keyword argument name for these objects so we can say that in this deprecation message?  Options:

- `result_type`
- `output_type`
- `result`
- `output_result`
- others?

**cc:** @lauralorenz

Reply:
Hmmm I don't love anything with `output`. Colloquially I feel it's too attached to stuff in stdout. I like `result` or `result_type`, the former if it is a result instance and the latter only if it is a reference to the class. I think we are pretty committed to the former.

Some other brainstorms... `persistence` or `persistence_type`; `return_type` (in the sense its kinda like type annotations for tasks, though this seems too narrow), `file` (feels basic but kinda true)

Reply:
I'd vote `result` for an instance and `result_cls` for the class to be consistent with the existing nomenclature (e.g. `runner_cls`, `executor_cls`, etc).

Reply:
I'm game for `result`!  In that case, can we do the following:

- update the warning message to include a mention of this new keyword
- actually introduce the keyword on both Tasks and Flows, but don't use it for anything yet (we can specify in the docstring "currently unused" until it is actually used by something)